harness/determined/keras/_tf_keras_trial.py

```diff
@@ -596,29 +589,78 @@ def run(self) -> None:
             self._launch_fit()
         except det.errors.WorkerFinishedGracefully:
             pass
+        finally:
+            self._stop_enqueuers()
 
     def _launch_fit(self) -> None:
-        (
-            training_input,
-            batches_per_epoch,
-        ) = self._train_input_manager.get_training_input_and_batches_per_epoch()
+        training_data = self.training_data
+
+        if isinstance(training_data, keras.SequenceAdapter):
```

Comment:
non-blocking: If this is only for backwards compatibility can we leave a comment saying so